File: Desktop/HomeWork2017/fib.py
class Fibonacci(object):
	def __init__(self):
		self.fib0 = 1
		self.fib1 = 1
		self.count = 0

	def __next__(self):
		if self.count == 0 or self.count == 1:
			self.count+=1
			return 1

		self.fib1 = self.fib0
		nextFib = self.fib0 + self.fib1
		self.fib0 = nextFib
		count+=1
		return self.fib1

	def __iter__():
		return self

def main():
	sequence = Fibonacci()
	for fib in range(10):
		print(next(fib))

if __name__ == '__main__':
	main()

File: Desktop/progs/ScalaCode/fib.scala
class fib(){
	def fib(n: Int) : Long = {
		if(n==1 || n==2) return 1;
		var t1 : Long = 1;
		var t2 : Long = 1;
		var next : Long = 0;
		for( x <- 3 to n+1) {
			next = t1+t2;
			t2 = t1;
			t1 = next;
		}
		next;
	}
	def apply() : fib = new fib();
}

object fib {
	def main(args: Array[String]): Unit = {
		val fib = new fib();
		println(fib.fib(3));
	}
}